guvnor-ala/guvnor-ala-registry-vfs/src/main/java/org/guvnor/ala/registry/vfs/VFSBuildRegistry.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.registry.vfs;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.guvnor.ala.registry.impl.BaseBuildRegistry;
+
+/**
+ * Builds are not persisted by now.
+ */
+@ApplicationScoped
+public class VFSBuildRegistry
```

Comment:
1) all the VFSXXXRegistry extends the BaseXXXRegistry counterpart.

2) the VFSXXXRegistries only override the required methods e.g. register(XXX) that is the place where we really want to write on the VFS. But they also want's to inherit certain behaviors from the parent class. Since It'll become inefficient to do all registry operations by reading all the elements form the VFS every time. 

3) I have some VFSXXXRegistries that still don't write in the VFS because we evaluated to write only some of them. But we might want to register more things in the VFS in next iteration.
Additionally since the registries are automatically plugable by CDI I don't want to package the guvnor-ala-registry-vfs and the guvnor-ala-registry-local at the same time. If I do this, I need to start adding @Specialized to some of the guvnor-ala-registry-vfs beans, and package two jars guvnor-ala-registry-vfs.jar and guvnor-ala-registry-local.jar in the generated webapp. And this can be confusing and bad looking.